### ISSUE TITLE
code generation for TypeScript about JsonExtensionData 

### ISSUE THREAD

**Issue by elementuse** [RSuter/NSwag#1541]
Is JsonExtensionData code generation not supported in typescript?

**Comment by elementuse** [comment]
in typescript.
is this code generator support?

**Comment by RSuter** [comment]
Yes, but you need something like this in typescript?

**Comment by elementuse** [comment]
yes. using typescript, my data  was filtered out by the proxy class.

**Comment by RSuter** [comment]
Yep, not implemented yet...

**Comment by elementuse** [comment]
How to avoid this problem and let my data display?
is there any good way?

**Comment by RSuter** [comment]
I think the only way to fix that right now is to use interfaces instead of classes for DTOs...

**Comment by Sharparam** [comment]
What's the status on this?

Not sure if it belongs here or in a new issue, but the code in the generated TypeScript code to handle `JsonExtensionData` doesn't seem to follow spec. It expects there to be a property "extensions" in the JSON data but that will not be there in correctly generated JSON, as specified in the documentation for the property `Extensions` on `ProblemDetails`:

> Extension members appear in the same namespace as other members of a problem type.

So for example, given this constructed `ProblemDetails`:

```csharp
new ProblemDetails
{
	// ...
	Extensions =
	{
		["CustomExtension"] = "custom value"
	}
};
```

The following JSON will be generated (standard fields omitted):

```json
{
	"CustomExtension": "custom value"
}
```

However, NSwag's TypeScript code expects this:

```json
{
	"extensions": {
		"CustomExtension": "custom value"
	}
}
```

Not sure how the C# JSON libraries handle them, but I suppose the way to handle it is to treat any root-level field on the JSON that doesn't have a matching property in the TypeScript class/interface as an extension and stuff it into the "extensions" property on the TypeScript class.
